### PR TITLE
fix: wrong gas limit when adjust base fee

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #823

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: go.mod
Comment (yihuang) [inline go.mod:159]: TODO: update after https://github.com/cosmos/cosmos-sdk/pull/10725 merged and backported to 0.44.x.

Comment (fedekunze) [reply go.mod:159]: isn't this a consensus change?

**Review by fedekunze — APPROVED** [review]
ACK, we should wait until it's backported to merge tho

File: go.mod
Comment (yihuang) [reply go.mod:159]: UPDATE: it's merged in release/v0.44.x, but not tagged yet.

**Comment by yihuang** [comment]
> ACK, we should wait until it's backported to merge tho

it's merged now, although not tagged yet.

**Comment by yihuang** [comment]
integration test failure is due to this issue: https://github.com/tharsis/ethermint/pull/851

**Review by fedekunze — APPROVED** [review]

**Pull request merged**
